Initial project structure. (#1)
php-fig/per-attributes: CONTRIBUTING.md

# Contributing

Contributions are **welcome** and will be fully **credited**.

We accept contributions via Pull Requests on [Github](https://github.com/php-fig/per-attributes).

## Pull Requests

- **[PER-Coding Standards](https://www.php-fig.org/per/coding-style/)** - Check the code style with `$ composer cs` and fix it with `$ composer cs-fix`.

- **Add tests!** - Your patch won't be accepted if it doesn't have tests.

- **Document any change in behaviour** - Make sure the `README.md` and any other relevant documentation are kept up-to-date.

- **Consider our release cycle** - We try to follow [SemVer v2.0.0](http://semver.org/). Randomly breaking public APIs is not an option.

- **Create feature branches** - Don't ask us to pull from your main branch.

- **One pull request per feature** - If you want to do more than one thing, send multiple pull requests.

> - **Send coherent history** - Make sure each individual commit in your pull request is meaningful. If you had to make multiple intermediate commits while developing, please [squash them](http://www.git-scm.com/book/en/v2/Git-Tools-Rewriting-History#Changing-Multiple-Commit-Messages) before submitting.

Reviewer: ```suggestion
- **Send coherent history** - Make sure each individual commit in your pull request is meaningful and atomic. Make sure each commit-message not only explains the **what** but also the **why**! 
```

Commits should be atomic. If there are multiple commits necessary they should be preserved to understand the process. The "Multiple intermediate commits" sounds like there should only be **one** commit per PR which I would rather not have people to get the impression of. They should use atomic commits and preserve them. 

And if we want meaningful commit-messages it is unfair towards the contributor to then squash them into one.

Author: This is inherited all the way from the PHP League template, I think.  I've barely looked at it in years.  Follow ups to change this part are fine by me, I don't have strong feelings about it, other than we give people good guidance.
